state/execution.go

```diff
@@ -118,9 +118,13 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 
 	// Fetch a limited amount of valid txs
 	maxDataBytes := types.MaxDataBytes(maxBytes, evSize, state.Validators.Size())
-	maxReapBytes := maxDataBytes
+	var maxReapBytes int64
 	if emptyMaxBytes {
 		maxReapBytes = -1
+	} else {
```

Comment:
nitpick: IMHO it's worth noting (in comment) that implementation of PrepareProposal (in app) respects the proper block size limit, so passing more transactions will not result in too big block.